keyboards/preonic/keymaps/trigotometry/keymap.c

```diff
@@ -0,0 +1,204 @@
+},
+
+/* Raise
+ * ,-----------------------------------------------------------------------------------.
+ * |  F12 |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |  F7  |  F8  |  F9  |  F10 |  F11 |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |  \   |      |      |   /  |      |      |      |      |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |  |   |  [   |  {   |  (   |  <   |   >  |   )  |   }  |   ]  |   |  |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |      |      |      |  +   |  -   |   _  |   =  |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |Dvorak|Qwerty| Game | Reset|
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = {
+  {KC_F12,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11 },
+  {_______, XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS, XXXXXXX, XXXXXXX, KC_PSLS, XXXXXXX, XXXXXXX, XXXXXXX, _______},
+  {_______, KC_PIPE, KC_LBRC, KC_LCBR, KC_LPRN, KC_LABK, KC_RABK, KC_RPRN, KC_RCBR, KC_RBRC, KC_PIPE, XXXXXXX},
+  {_______, XXXXXXX, XXXXXXX, XXXXXXX, KC_PPLS, KC_PMNS, KC_UNDS, KC_EQL,  XXXXXXX, XXXXXXX, XXXXXXX, _______},
+  {_______, _______, _______, _______, _______, _______, _______, _______, DVORAK,  QWERTY,  GAMING,    RESET}
+}
+
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+
+        case DVORAK:
+          if (record->event.pressed) {
+            persistent_default_layer_set(1UL<<_DVORAK);
+          }
+          return false;
+          break;
+          
+        case QWERTY:
+          if (record->event.pressed) {
+            persistent_default_layer_set(1UL<<_QWERTY);
+          }
+          return false;
+          break;
+
+        case LOWER:
+          if (record->event.pressed) {
+            layer_on(_LOWER);
+          } else {
+            layer_off(_LOWER);
+          }
+          return false;
+          break;
+
+        case RAISE:
+          if (record->event.pressed) {
+            layer_on(_RAISE);
+          } else {
+            layer_off(_RAISE);
+          }
+          return false;
+          break;
+
+        case GAMING:
+          if (record->event.pressed) {
+            persistent_default_layer_set(1UL<<_GAMING);
+          }
+          return false;
+          break;
+
+      }
+    return true;
+};
+
+#ifdef AUDIO_ENABLE
+float tone_startup[][2] = {
+  {NOTE_D5, 8},
+  {NOTE_E6, 8},
+  {NOTE_D6, 8},
+  {NOTE_E5, 8}
+};
+
+void startup_user()
```

Comment:
This doesn't actually get called by anything. 

Additionally, if you use `#define STARTUP_SONG SONG(STARTUP_SOUND)` in your config.h file, it will play this.

Reply:
Well, the above may handle that, as well. 

But yeah, if your keymap doesn't call the `startup_user();` function, nothing else in QMK does.

Reply:
I'm only just getting started with learning Python so I'm totally a rookie here to the point that I assumed it was being called.

I'll do my best with the rest of the changes and try again! Appreciate your patience and time mate.

Reply:
Well, QMK is very modular, so it's easy to get confused if you're not used to it.  Which is why I mentioned this.

That said, I would recommend trying the startup sound as mentioned above, and see if that works.

Also, is this an ARM board (eg, the rev3)? I'm pretty sure the answer is no, but just wanted to make sure.

Reply:
I'll give it a go in my next Pull.

Nah it's a Rev2.

Reply:
Okay, sounds good! 

Is this ready to go?

Reply:
Almost, doing a quick ReadMe for the fans. Heh.